File: Array/1.Easy/01.Largest_element_in_array.cpp
/*
QUESTION:-
Given an array A[] of size n. The task is to find the largest element in it.

Example:

Input:
n = 5
A[] = {1, 8, 7, 56, 90}
Output:
90
Explanation:
The largest element of given array is 90
*/

/*
APPROACH:-
-> Intialize the ans with starting element
-> Traverse the entire array and update the ans if the element is greater then ans
-> Finally, return the ans
*/

#include <bits/stdc++.h>
using namespace std;

// CODE:-
int largest(int arr[], int n)
{
    int ans = arr[0];
    for (int i = 1; i < n; i++)
    {
        if (arr[i] > ans)
            ans = arr[i];
    }
    return ans;
}

int main() {
    // Test case 1
    cout << "Test 1:" << endl;
    int n1 = 5;
    int arr1[] = {1, 8, 7, 56, 90};
    int result1 = largest(arr1, n1);
    cout << "Output: " << result1 << endl;
    cout << "Expected: 1" << endl;
    cout << endl;

    return 0;
}

// TIME COMPLEXITY = O(N)

File: 01.Arrays/1.Easy/01.Largest_element_in_array.py
# QUESTION:-
# Given an array A[] of size n, find the largest element in it.
# Example: Input: n = 5, A[] = {1, 8, 7, 56, 90}, Output: 90

# APPROACH:-
# Initialize ans with the first element
# Traverse the entire array and update ans if element is greater
# Finally, return ans

# CODE:-
def largest(arr, n):
    """Find the largest element in array"""
    ans = arr[0]
    for i in range(1, n):
        if arr[i] > ans:
            ans = arr[i]
    return ans


if __name__ == '__main__':
    # Test case 1
    print("Test 1:")
    n1 = 5
    arr1 = [1, 8, 7, 56, 90]
    result1 = largest(arr1, n1)
    print(f"Output: {result1}")
    print(f"Expected: 90")

# TIME COMPLEXITY = O(N) 
